Create ssl-https.md (#173)
NoroffFEU/development-platforms: src/content/docs/webdev-hosting/https-ssl/ssl-https.md

---
title: SLL/HTTPS Case Study

> author: Erlend Espe

Reviewer: You can add your github username to the author field too

## HTTPS
HTTPS is the secure version of its predecessor HTTP. HTTP is the primary protocol used to send data between a web browser
and a website. HTTPS does the same thing, but it encrypts the data in order to improve security.
HTTPS uses an encryption protocol to secure communications. This protocol is called Transport Layer Security(TLS), previously known as SSL. TLS secures communications by using an asymmetric public key infrastructure. This means it uses two different keys to encrypt data sent between two parties.
- The private key: This key is controlled by the owner of a website and is kept private. This key is stored on a web server and is used to decrypt information which is encrypted by the public key.
- The public key: This key is not private, and is available to anyone that wants to interact with the server in a secure way.
When a user accesses a webpage, the webpage transmits its SSL certificate, containing the public key for initiating a secure session. The client and server undergo an SSL/TLS handshake, a sequence of exchanges designed to establish a secure connection.

## Benefits of encryption

> HTTPS prevents data sent from a user to a webserver, or opposite, from being openly viewed by anyone snooping on the network. Data sent over regular HTTP is broken into packets of data which can easily be sniffed by using software like wireshark. Using HTTPS is particularly important when using public wifi, because anyone connected to this wifi can view your data in plain text, and makes you particularly vulnerable to on-path attacks etc.

Reviewer: ```suggestion
HTTPS prevents data sent from a user to a webserver, or the opposite, from being openly viewed by anyone snooping on the network. Data sent over regular HTTP is broken into packets of data which can easily be sniffed by using software like Wireshark. Using HTTPS is particularly important when using public wifi, because anyone connected to this wifi can view your data in plain text, and makes you particularly vulnerable to on-path attacks etc.
```

When using HTTPS, data is encrypted and will come across as nosence characters. Example:

> ![image](https://github.com/ErlendE96/devplat/assets/96021420/945cd91f-1e8b-4edc-830c-94be41d0d155)

Reviewer: The image needs to be relative to this repo and be included in the PR.

Websites not using HTTPS are vulnerable to internet service providers or other intermediaries to inject content into webpages without the approval of the owner of the website.The content injected is commonly advertising, which typically reduces the quality of the adds and the generated revenue is not shared with the website owner.
Bank details, login information and other sensitive information is also vulnerable when not using HTTPS.

## HTTP/HTTPS
technically HTTPS is not a seperate protocol from HTTP, it is using TLS or SSL encryption over the HTTP protocol.

## Weaknesses

> HTTPS encrypts the entire HTTP request and response, but the DNS resolution and connection setup can reveal some information, like for example the full domain or subdomain and the originating IP adress.

Reviewer: ```suggestion
HTTPS encrypts the entire HTTP request and response, but the DNS resolution and connection setup can reveal some information, like for example the full domain or subdomain and the originating IP address.
```

Attackers are also able to analyze encrypted traffic for side channel information, such as time spent on site and the size of user input.

## Setting up HTTPS

> In order to use HTTPS with your domain name, a SSL or TLS sertificate is needed to be installed on the website. Your webhost may offer HTTPS security or a SSL/TLS sertificate may be requested from Certificate Authoroties and installed by yourself. There are multiple types of webservers and each has its own process for installing and updating SSL/TLS certificates.

Reviewer: ```suggestion
In order to use HTTPS with your domain name, a SSL or TLS sertificate is needed to be installed on the website. Your webhost may offer HTTPS security or a SSL/TLS sertificate may be requested from Certificate Authorities and installed by yourself. There are multiple types of web servers and each has its own process for installing and updating SSL/TLS certificates.
```
